wire-ios-request-strategy/Sources/Request Strategies/Assets/AssetClientMessageRequestStrategy.swift

```diff
@@ -60,16 +60,27 @@ extension AssetClientMessageRequestStrategy: InsertedObjectSyncTranscoder {
     typealias Object = ZMAssetClientMessage
 
     func insert(object: ZMAssetClientMessage, completion: @escaping () -> Void) {
+        let logAttributesBuilder = MessageLogAttributesBuilder(context: managedObjectContext)
+        let logAttributes = logAttributesBuilder.syncLogAttributes(object)
+        WireLogger.messaging.debug("inserting message", attributes: logAttributes)
+
         // Enter groups to enable waiting for message sending to complete in tests
         let groups = managedObjectContext.enterAllGroupsExceptSecondary()
         Task {
             do {
                 try await messageSender.sendMessage(message: object)
+
+                let logAttributes = await logAttributesBuilder.logAttributes(object)
```

Comment:
Not a request for change here, but something to think about. In general I don't think logging should block work being done because it can kind of become a hidden dependency of behavior. However I'm not sure how you could do things differently 🤔

Reply:
yes, here it's just we want to log info from managedObjects that are on a specific context/thread.